Explorer/Assets/DCL/Landscape/TerrainGenerator.cs

```diff
@@ -0,0 +1,520 @@
+            ocean.SetParent(rootGo.transform, true);
+
+            Transform wind = Object.Instantiate(terrainGenData.wind).transform;
+            wind.SetParent(rootGo.transform, true);
+        }
+
+        //           size,size
+        //      N
+        //    W + E
+        //      S
+        //0,0
+
+        private void GenerateCliffs()
+        {
+            if (terrainGenData.cliffSide == null || terrainGenData.cliffCorner == null)
+                return;
+
+            CreateCliffCornerAt(new Vector3(terrainGenData.terrainSize, 0, terrainGenData.terrainSize), Quaternion.identity);
+            CreateCliffCornerAt(new Vector3(terrainGenData.terrainSize, 0, 0), Quaternion.Euler(0, 90, 0));
+            CreateCliffCornerAt(new Vector3(0, 0, 0), Quaternion.Euler(0, 180, 0));
+            CreateCliffCornerAt(new Vector3(0, 0, terrainGenData.terrainSize), Quaternion.Euler(0, 270, 0));
+
+            for (var i = 0; i < terrainGenData.terrainSize; i += 16)
+            {
+                Transform side = Object.Instantiate(terrainGenData.cliffSide).transform;
+                side.position = new Vector3(terrainGenData.terrainSize, 0, i + 16);
+                side.rotation = Quaternion.Euler(0, 90, 0);
+                side.SetParent(rootGo.transform, true);
+            }
+
+            for (var i = 0; i < terrainGenData.terrainSize; i += 16)
+            {
+                Transform side = Object.Instantiate(terrainGenData.cliffSide).transform;
+                side.position = new Vector3(i, 0, terrainGenData.terrainSize);
+                side.rotation = Quaternion.identity;
+                side.SetParent(rootGo.transform, true);
+            }
+
+            for (var i = 0; i < terrainGenData.terrainSize; i += 16)
+            {
+                Transform side = Object.Instantiate(terrainGenData.cliffSide).transform;
+                side.position = new Vector3(0, 0, i);
+                side.rotation = Quaternion.Euler(0, 270, 0);
+                side.SetParent(rootGo.transform, true);
+            }
+
+            for (var i = 0; i < terrainGenData.terrainSize; i += 16)
+            {
+                Transform side = Object.Instantiate(terrainGenData.cliffSide).transform;
+                side.position = new Vector3(i + 16, 0, 0);
+                side.rotation = Quaternion.Euler(0, 180, 0);
+                side.SetParent(rootGo.transform, true);
+            }
+        }
+
+        private void CreateCliffCornerAt(Vector3 position, Quaternion rotation)
+        {
+            Transform neCorner = Object.Instantiate(terrainGenData.cliffCorner).transform;
+            neCorner.position = position;
+            neCorner.rotation = rotation;
+            neCorner.SetParent(rootGo.transform, true);
+        }
+
+        private async UniTask SetupEmptyParcelDataAsync()
+        {
+            emptyParcelResult = new NativeHashMap<int2, EmptyParcelData>(emptyParcels.Length, Allocator.Persistent);
```

Comment:
I think it can have a job allocator, you don't need it to be persistent, do you?

Reply:
It needs to be persistent since its being used on multiple jobs